Add and update wiki pages, add Thunder download. (#90)
GeyserMC/GeyserWebsite: wiki/other/thunder.md

---
title: Thunder
description: 'A java application to convert simple Java Edition resource packs to Bedrock Edition ones.'
---

Thunder is a java application to convert simple Java Edition resource packs to Bedrock Edition ones.

## What is Thunder? {#what-is-thunder}

Thunder is a converter for simple Java Edition resource packs which can generate Bedrock Edition packs, it currently converts resource packs which modifies existing vanilla assets.

:::caution

Not only is this project in early beta, it does **not** convert custom items! Please look towards [Rainbow](/wiki/other/rainbow) for that.

:::

## Usage {#usage}

> 1. Ensure you have Java installed, if you do you should be able to open the file like a normal program.

Reviewer: java 17

> 2. Select your pack using the big button on the left. Once you do, if it was successful you should see the icon of the pack has showed up.

Reviewer: ```suggestion
2. Select the Java edition resource pack using the button on the left. Once you do, if it was successful you should see the icon of the pack.
```

> 3. Change the output pack name to what you desire, this changes the name shown in bedrock, not the file name.

Reviewer: ```suggestion
3. Change the output pack name. This is the name shown to the Bedrock client.
```

> 4. Hit the convert button and be patient while it converts, it may take a while.

Reviewer: ```suggestion
4. Hit the convert button to start the conversion process. This step may take a while.
```

> 5. Once completed, your pack should be put in the same place as where your Java pack was, you can now either use the pack on your client if your device supports adding custom packs, or place it in the `packs` folder in your Geyser server.

Reviewer: ```suggestion
5. Once completed, the converter Bedrock edition resource pack will be in the same place as where your Java pack was. 
You can now either use the pack on the Bedrock client, or place it in the `packs` folder in your Geyser server.
```

## Download {#download} 

You can download Thunder [here](/download/?project=other-projects&thunder=expanded).

## Contributing {#contributing}

Any contributions are appreciated. Please feel free to reach out to us on [Discord](https://discord.gg/geysermc) if
you're interested in helping out with Thunder.

### Project Setup {#project-setup}

1. Clone the repo to your computer

> 2. Run gradlew build and locate to bootstrap/build folder.

Reviewer: Same as above, I'd yeet this here. Maybe worth adding a "for developers" section that mentions that thunder is just a gui for the underlying pack converter library

Author: Yeeted, although, I'm not sure a "for developers" section makes the most sense for a sentence or 2, perhaps on each page we have a `Source Code` link to get people going to the GitHub repo for it, there would be a more appropriate place to have information like that
